File: week-04/day-4/01.py
# 1. write a recursive function
# that takes one parameter: n
# and counts down from n

def countdown(n):
    if n <= 0:
        pass
    else:
        print(n)
        countdown(n-1)
countdown(10)


# def count(n):
#     for i in range(n, 0, -1):
#         print(i)
# count(10)

# def concat(n):
#     if n <= 0:
#         return []
#     else:
#         return concat(n-1) + [n]
# print(concat(3))

File: week-08/day-3/01.js
'use strict';

// create a function that takes a filename reads the content of the file
// and counts the words in it. It should not return the result, rather
// call a callback (its second parameter)
// The callback should have two parameters:
//  - err: the error object if anything wrong happened
//  - count: the word count

var fs = require('fs');


function wordCount(fileName, cb) {
  fs.readFile(fileName, function(err, content){
    if (err) {
      return cb(err);
    }
    var count = String(content).replace('/\r?\n|\r/g', ' ').split(' ').length-1;
    cb(null, count);
  });
}

wordCount('alma.txt', function(err, count) {
  console.log(count);
})
